src/NLog/Layouts/XmlElementBase.cs

```diff
@@ -0,0 +1,760 @@
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution. 
+// 
+// * Neither the name of Jaroslaw Kowalski nor the names of its 
+//   contributors may be used to endorse or promote products derived from this
+//   software without specific prior written permission. 
+// 
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE 
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE 
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE 
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR 
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN 
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) 
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF 
+// THE POSSIBILITY OF SUCH DAMAGE.
+//
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Text;
+using NLog.Config;
+using NLog.Internal;
+
+namespace NLog.Layouts
+{
+    /// <summary>
+    /// A specialized layout that renders XML-formatted events.
+    /// </summary>
+    [ThreadAgnostic]
+    [ThreadSafe]
+    public abstract class XmlElementBase : Layout
+    {
+
+        private const string DefaultPropertyName = "property";
+        private const string DefaultPropertyKeyAttribute = "key";
+        private const string DefaultCollectionItemName = "item";
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="XmlElementBase"/> class.
+        /// </summary>
+        /// <param name="elementName">The name of the top XML node</param>
+        /// <param name="elementValue">The value of the top XML node</param>
+        protected XmlElementBase(string elementName, Layout elementValue)
+        {
+            ElementNameInternal = elementName;
+            ElementValueInternal = elementValue;
+            Attributes = new List<XmlAttribute>();
+            Elements = new List<XmlElement>();
+            ExcludeProperties = new HashSet<string>();
+        }
+
+        /// <summary>
+        /// Name of the XML element
+        /// </summary>
+        /// <docgen category='XML Options' order='10' />
+        protected string ElementNameInternal { get => _elementName; set => _elementName = XmlHelper.XmlConvertToElementName(value?.Trim(), true); }
+        private string _elementName;
+
+        /// <summary>
+        /// Value inside the XML element
+        /// </summary>
+        /// <docgen category='XML Options' order='10' />
+        protected Layout ElementValueInternal { get => _elementValueWrapper.Inner; set => _elementValueWrapper.Inner = value; }
```

Comment:
Consider making this internal instead of protected.

Reply:
done (with remark note added), thanks!